### PR TITLE
Use same logic to get gvk of resource object

### PR DESCRIPTION
**Pull request by elgnay**
This is fllowup of https://github.com/open-cluster-management/work/pull/15/files#r428286111.

### CODE REVIEW

File: pkg/spoke/controllers/manifestwork_controller.go [hunk 1]
```diff
@@ -411,48 +411,28 @@ func buildManifestResourceMeta(index int, object runtime.Object, restMapper *res
 		Ordinal: int32(index),
 	}
 
-	if object == nil {
+	if reflect.ValueOf(object).IsNil() {
```
Comment (qiujian16) [inline pkg/spoke/controllers/manifestwork_controller.go]: I think we still need to check if object==nil before this? I think reflect.ValueOf(nil) will panic.

Comment (elgnay) [reply pkg/spoke/controllers/manifestwork_controller.go]: Yes, you are right. Code is updated.

**Pull request merged**